pkg/es/config/config.go

```diff
@@ -329,12 +339,17 @@ func (tlsConfig *TLSConfig) loadPrivateKey() (*tls.Certificate, error) {
 
 // TokenAuthTransport
 type tokenAuthTransport struct {
-	token   string
-	wrapped *http.Transport
+	token                string
+	allowOverrideFromCtx bool
+	wrapped              *http.Transport
 }
 
 func (tr *tokenAuthTransport) RoundTrip(r *http.Request) (*http.Response, error) {
-	r.Header.Set("Authorization", "Bearer "+tr.token)
+	token := tr.token
+	if tr.allowOverrideFromCtx {
+		token, _ = spanstore.GetBearerToken(r.Context())
+	}
+	r.Header.Set("Authorization", "Bearer "+token)
```

Comment:
You need some checks here. Previously, the token came from a local file, which means that someone trusted placed it there. 

The source of data is now untrusted, as a malicious user could send a especially crafted token and it would be relayed to the next server (Elasticsearch, in this case), similar to the Smuggling Attack.

Check if there's a library in Go that you can make use of to sanitize this input.

Reply:
Is there a way to validate a bearer token? I'm asking because it suppose to be an opaque string,  type of access token, it means that there is no specific format for it. But I'll do an investigation on to see what things we should sanitize here.

Reply:
Bearer token can be anything, even though it's typically a JWT. But as it's an HTTP header, you can sanitize based on the general HTTP header rules.

Reply:
@jpkrohling  http golang package parse the headers, not sure but I'm assuming at that point it should be a valid header. I haven't found anything about header sanitization.

Reply:
Actually, scratch that: your source is already an HTTP header itself, so, what you have is already a valid HTTP header content. If you were receiving this as a form parameter, you'd have to sanitize it, though.